### PR TITLE
[ENG-2790] Fix broken p0 kubeconfig command

### PR DESCRIPTION
**Pull request by p0-andrewa**
This PR unbreaks the `p0 kubeconfig` command. Some values and structures got changed around in the k8s integration config and responses from the backend:
- Required AWS IDC metadata is no longer found in the `eksGenerated` object; it's now in `awsResourcePermission`
- AWS account ID is no longer stored separately in the k8s integration configuration, and now is extracted from the EKS cluster ARN
- `provider` was renamed `hosting` in the k8s integration configuration

It also adds some minor quality-of-live improvements to the `p0 kubeconfig` command, such as utilizing `spinUntil()` for some of the longer `await`s, and adds a more robust and reusable simple AWS ARN parser.

Added a unit test for the ARN parser, and tested everything locally in both AWS IDC and AWS Federated (Okta SAML) configurations.

### CODE REVIEW

File: src/plugins/kubeconfig/index.ts
Comment (gergas3) [inline src/plugins/kubeconfig/index.ts:117]: Why can the isPreexisting be removed?

**Review by gergas3 — APPROVED** [review]
🙏 Thank you for the fix.
lgtm with one question

File: src/commands/kubeconfig.ts
Comment (nbrahms) [inline src/commands/kubeconfig.ts:243]:
```suggestion
  const [resourceType, clusterName] = resourceStr.split("/");
```

🤷

File: src/plugins/aws/utils.ts
Comment (nbrahms) [inline src/plugins/aws/utils.ts:68]: With
```ts
const ARN_PATTERN =
  /^arn:(aws|aws-us-gov|aws-cn|aws-iso|aws-iso-b):([^:]*):([^:]*):([^:]*):(.*)$/;
```

```suggestion
  const invalidArnMessaage = `Invalid AWS ARN: ${arn}`;
  const match = arn.match(ARN_PATTERN);
  
  if (!match) {
    throw INVALID_ARN_MSG;
  }
  
  const [partition, service, region, accountId, resource] = match;

  return {
    partition,
    service,
    region,
    accountId,
    resource,
  };
```

File: src/plugins/kubeconfig/index.ts
Comment (nbrahms) [reply src/plugins/kubeconfig/index.ts:117]: Isn't `isPreexisting` about approval, not provisioning? Seems like you want to print a "waiting for approval" message when requesting access only if `!preExisting`, but always print a "waiting for ... to be provisioned" whenever an access request is made.

**Review by nbrahms — APPROVED** [review]
See comments

File: src/plugins/kubeconfig/index.ts
Comment (p0-andrewa) [reply src/plugins/kubeconfig/index.ts:117]: Yes, I believe it's about the approval, not provisioning per se, so the original message wasn't really right either. Either way, the real reason for my removing this is that it was intended as a way to tell the user that the CLI isn't frozen, just that it's waiting for the request to complete, but this is now taken care of by the `spinUntil()` call below this.

File: src/plugins/aws/utils.ts
Comment (p0-andrewa) [reply src/plugins/aws/utils.ts:68]: This still shows the same `TS2322: Type string | undefined is not assignable to type string` error; I've adopted the change, but it looks like the `if (partition === undefined ...` block is still needed.

Comment (p0-andrewa) [reply src/plugins/aws/utils.ts]: Okay, this is working now with a small modification, but a question before I merge—do we actually want to enforce a value for the partition in the regex? AWS doesn't make any guarantees about what partitions exist, and as long as the ARN is correctly formatted, I'm not sure it makes sense to enforce that it's one of these specifically.

**Pull request merged**
